src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useCallback } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Box } from '@mui/material';
+import { Todo } from '../../types/Todo';
+import { TodoElement } from '../TodoElement';
+
+type Props = {
+  todos: Todo[];
+  onDelete: (id: number) => void;
+  onUpdateTodo: (id: number, data: Partial<Todo>) => void;
+  tempTodo: Todo | null,
+  isTodoLoading: (id: number) => boolean,
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  onDelete,
+  onUpdateTodo,
+  tempTodo,
+  isTodoLoading,
+}) => {
+  const getTodos = useCallback((filterType: string): Todo[] => {
+    switch (filterType) {
+      case 'active':
+        return todos.filter(todo => !todo.completed);
+
+      case 'completed':
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [todos]);
+
+  const { pathname } = useLocation();
+
+  const visibleTodos = getTodos(pathname.slice(1));
```

Comment:
Do all the stuff with a location in the App component. It will be your one source of truth. Then you can pass the filter type to children as props.